### PR TITLE
Segmentation on Clay

### PR DESCRIPTION
**Pull request by srmsoumya**
Add segmentation support for Clay
- Head from Segformer paper: https://arxiv.org/abs/2105.15203
- Tested on a subset of Chesapeake Bay CVPR dataset with 7 classes

### CODE REVIEW

**Review by yellowcap — CHANGES_REQUESTED** [review]
Could you add some docs about how to download the Chesapeake dataset to fully run this independently?

File: finetune/segment/chesapeake_model.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+from finetune.segment.factory import Segmentor
+
+
+class ChesapeakeSegmentor(L.LightningModule):
+    def __init__(
+        self,
+        metadata_path="configs/metadata.yaml",
+        num_classes=7,
+        feature_maps=[3, 5, 7, 11],
+        ckpt_path="checkpoints/v0.5.7/mae_v0.5.7_epoch-13_val-loss-0.3098.ckpt",
```
[the remark this answers is not in the document]
Comment (srmsoumya) [reply finetune/segment/chesapeake_model.py:19]: Done.

File: configs/segment_chesapeake.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  platform: naip
+model:
+  metadata_path: configs/metadata.yaml
+  num_classes: 7
+  feature_maps:
+    - 3
+    - 5
+    - 7
+    - 11
+  ckpt_path: checkpoints/v0.5.7/mae_v0.5.7_epoch-13_val-loss-0.3098.ckpt
```
[the remark this answers is not in the document]
Comment (srmsoumya) [reply configs/segment_chesapeake.yaml:20]: Done.

File: configs/segment_chesapeake.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  logger:
+    - class_path: lightning.pytorch.loggers.WandbLogger
+      init_args:
+        entity: developmentseed
+        project: clay-segment
+        log_model: false
+  callbacks:
+    - class_path: lightning.pytorch.callbacks.ModelCheckpoint
+      init_args:
+        dirpath: s3://clay-model-ckpt/finetune/segment/chesapeake/v0.0.1/
```
[the remark this answers is not in the document]
Comment (srmsoumya) [reply configs/segment_chesapeake.yaml:43]: Same as above.

File: finetune/segment/README.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+```bash
+git clone <repo-url>
+cd model
+mamba env create --file environment.yml
+mamba activate claymodel
+```
+
+## Usage
+
+### Preparing the Dataset
```
Comment (srmsoumya) [inline finetune/segment/README.md:44]: @yellowcap we have details on how to prepare the dataset for segmentation task.

File: finetune/segment/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        - 7
+        - 11
+      ckpt_path: checkpoints/clay-v1-base.ckpt
+      lr: 1e-5
+      wd: 0.05
+      b1: 0.9
+      b2: 0.95
+    ```
+
+3. Update the WandB logger configuration in the configuration file with your WandB details:
```
Comment (yellowcap) [inline finetune/segment/README.md:103]: Could we add instructions for folks that don't have wandb accounts?

**Review by yellowcap — APPROVED** [review]
One small comment, otherwise looks good.

**Pull request merged**